k8s/elasticsearch/apptest/tester/tests/basic-suite.yaml

```diff
@@ -1,8 +1,89 @@
 actions:
-- name: dummy test
+- name: Wait until all expected nodes are connected to the cluster
   bashTest:
     script: |-
-      exit 0
+      function get_nodes_count() {
+        curl -s "${HEALTH_URL}" | jq '.number_of_nodes'
+      }
+
+      expected_nodes="${REPLICAS}"
+
+      timeout=300
+      timeout_time=$(( $(date +%s) + timeout ))
+
+      # Wait for all nodes to join the cluster
+      nodes_count="$(get_nodes_count)"
+      while [[ "${nodes_count}" -ne "${expected_nodes}" ]] \
+          && [[ "$(date +%s)" -le "${timeout_time}" ]]; do
+        sleep 3
+        nodes_count="$(get_nodes_count)"
+      done
+
+      if [[ "${nodes_count}" -eq "${expected_nodes}" ]]; then
+        echo "OK - all nodes joined"
+      else
+        echo "FAIL - nodes count: ${nodes_count}"
+      fi
+    expect:
+      stdout:
+        contains: OK - all nodes joined
+
+- name: Wait for green status in the cluster
+  bashTest:
+    script: |-
+      function get_cluster_status() {
+        curl -s "${HEALTH_URL}" | jq '.status'
+      }
+
+      timeout=300
+      timeout_time=$(( $(date +%s) + timeout ))
+
+      # Wait for green status
+      status="$(get_cluster_status)"
+      while [[ "${status}" != '"green"' ]] \
+          && [[ "$(date +%s)" -le "${timeout_time}" ]]; do
+        sleep 3
+        status="$(get_cluster_status)"
+      done
+
+      if [[ "${status}" == '"green"' ]]; then
+        echo "OK - green status"
+      else
+        echo "FAIL - status: ${status}"
+      fi
+    expect:
+      stdout:
+        contains: OK - green status
+
+- name: Add a document
```

Comment:
Can you name steps in a question form? `Can add a document` instead

Reply:
I rephrased the names.